itest/lnd_sweep_test.go

```diff
@@ -2150,3 +2152,114 @@ func runBumpFee(ht *lntest.HarnessTest, alice *node.HarnessNode) {
+	chanPoint := chanPoints[0]
+	alice := nodes[0]
+
+	// Alice force closes the channel which has no HTLCs at stake.
+	_, closingTxID := ht.CloseChannelAssertPending(alice, chanPoint, true)
+	require.NotNil(ht, closingTxID)
+
+	// Alice should see one waiting close channel.
+	ht.AssertNumWaitingClose(alice, 1)
+
+	// Alice should have 2 registered sweep inputs. The anchor of the local
+	// commitment tx and the anchor of the remote commitment tx.
+	ht.AssertNumPendingSweeps(alice, 2)
+
+	// Calculate the commitment tx fee rate.
+	closingTx := ht.AssertTxInMempool(closingTxID)
+	require.NotNil(ht, closingTx)
+
+	// The default commitment fee for anchor channels is capped at 2500
+	// sat/kw but there might be some inaccuracies because of the witness
+	// signature length therefore we calculate the exact value here.
+	closingFeeRate := ht.CalculateTxFeeRate(closingTx)
+	parentWeight := ht.CalculateTxWeight(closingTx)
+	parentFee := ht.CalculateTxFee(closingTx)
+
+	// Expect a 2 input 1 output transaction when bumping the force-close
+	// transaction. We have a large enough input because alice had a 1 BTC
+	// input to start with.
+	estimator := input.TxWeightEstimator{}
+	estimator.AddTaprootKeySpendInput(txscript.SigHashDefault)
+
+	// We are dealing with non-taproot channels here.
+	estimator.AddWitnessInput(input.AnchorWitnessSize)
+
+	// LND by default creates p2tr outputs.
+	estimator.AddP2TROutput()
+	childWeight := estimator.Weight()
+
+	// We increase the fee rate of the fee function by 100% to make sure
+	// we trigger a cpfp-transaction.
+	newFeeRate := closingFeeRate * 2
+
+	// We need to make sure that the budget can cover the fees for bumping
+	// the close transction taking also the parent fee into account.
+	// Because we are not taking the parent fee rate into account when
+	// crafting the initial cpfp transaction even less will be used but
+	// technically this is the minimum budget we need to bump the fee rate
+	// to the new starting fee rate of the fee function.
+	minBudget := newFeeRate.FeeForWeight(parentWeight+childWeight) -
+		parentFee
+
+	bumpFeeReq := &walletrpc.BumpForceCloseFeeRequest{
+		ChanPoint:   chanPoint,
+		SatPerVbyte: uint64(newFeeRate.FeePerVByte()),
+		// The budget needs to be high enough to pay for the fee because
+		// the anchor does not have an output value high enough to pay
+		// for itself.
+		Budget: uint64(minBudget),
+		// We use a force param to create the sweeping tx immediately.
+		Immediate: true,
+	}
+
+	alice.RPC.BumpForceCloseFee(bumpFeeReq)
+
+	// We expect the initial closing transaction and the local anchor cpfp
+	// transaction because alice force closed the channel.
+	// NOTE: We don't compare a feerate but only make sure that a cpfp
+	// transaction was triggered. The sweeper increases the fee rate
+	// periodically with every new incoming block and the selected fee
+	// function.
+	ht.AssertNumTxsInMempool(2)
+
+	// Mine both transactions, the closing tx and the anchor cpfp tx.
```

Comment:
Quite the opposite, that's why I mentioned we don't "need to check the feerate" or "check the specifics" here, we just check the sweeping attempt can be made.

I think it's important to understand what we are testing here. We are not re-testing the fee bumping logic again, either it's the feerate being increased or skipped. Instead, we are testing the new RPC endpoint, that a user can successfully use to bump the force close tx. There are many edge cases we could test, but I think the common one is,
1. force close happened
2. the user wants to fee bump it, but with a small budget, so calls the RPC.
3. the tx is not confirmed in the next block, the user wants to fee bump it again, so calls the RPC again.

We may also need to check that two versions of the anchor sweeping requests are made every time we call this RPC. And assert the sweeping tx is indeed using one of the anchors, not both - to ensure this new RPC won't prematurely remove the exclusive group.

As for neutrino, I think either we made a dedicated test for it (to avoid nested if-else) or we could skip it. I'm leaning towards skipping it.

Reply:
I reworked the itest, now instead of mining the mempool transaction immediately I make sure I call the `BumpForceCloseFee` rpc two times and make sure that both of the bumps use the same anchor input.

Did exclude the neutrino case.

What I am not really sure about is the following:

```
We may also need to check that two versions of the anchor sweeping requests are made every time we call this RPC. And assert the sweeping tx is indeed using one of the anchors, not both - to ensure this new RPC won't prematurely remove the exclusive group.
```

How can I check that actually two sweeps are attempted (local, remote) every time I bump the close fee ? Maybe you have an idea ?